hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/chillmode/ChillModeExitRule.java

```diff
@@ -17,16 +17,79 @@
  */
 package org.apache.hadoop.hdds.scm.chillmode;
 
+import org.apache.hadoop.hdds.server.events.EventHandler;
+import org.apache.hadoop.hdds.server.events.EventPublisher;
+
+
 /**
- * Interface for defining chill mode exit rules.
+ * Abstract class for ChillModeExitRules. When a new rule is added, the new
+ * rule should extend this abstract class.
+ *
+ * Each rule Should do:
+ * 1. Should add a handler for the event it is looking for during the
+ * initialization of the rule.
+ * 2. Add the rule in ScmChillModeManager to list of the rules.
+ *
  *
  * @param <T>
  */
-public interface ChillModeExitRule<T> {
```

Comment:
Just food for thought: You could actually still keep this as an interface and provide a default implementation for onMessage. New Java 8 contract change that avoids introducing inheritance purely for a default impl.

Reply:
Yes, it can be used but we don't want subclasses to implement this. So in the next patch, I made this method as final in abstract class. This came up when discussing with Nanda, so left this as an abstract class.

But your suggestion can be done, if we don't have such a requirement.